Assets/MixedRealityToolkit/InputSystem/Sources/GestureInputSource.cs

```diff
@@ -0,0 +1,191 @@
+    /// </summary>
+    public class GestureInputSource : BaseGenericInputSource
+    {
+        /// <summary>
+        /// Pointer Action for Gesture Tap or "Click"
+        /// </summary>
+        public static MixedRealityInputAction PointerAction { get; set; } = MixedRealityInputAction.None;
+
+        /// <summary>
+        /// Hold action to use when events are raised.
+        /// </summary>
+        public static MixedRealityInputAction HoldAction { get; set; } = MixedRealityInputAction.None;
+
+        /// <summary>
+        /// Manipulation action to use when events are raised.
+        /// </summary>
+        public static MixedRealityInputAction ManipulationAction { get; set; } = MixedRealityInputAction.None;
+
+        /// <summary>
+        /// Navigation action to use when events are raised.
+        /// </summary>
+        public static MixedRealityInputAction NavigationAction { get; set; } = MixedRealityInputAction.None;
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="useRailsNavigation">Should the gesture input source use rails navigation?</param>
+        public GestureInputSource(bool useRailsNavigation) : base("Gesture Input Source")
+        {
+            gestureRecognizer = new GestureRecognizer();
+
+            gestureRecognizer.Tapped += GestureRecognizer_Tapped;
+
+            gestureRecognizer.HoldStarted += GestureRecognizer_HoldStarted;
+            gestureRecognizer.HoldCompleted += GestureRecognizer_HoldCompleted;
+            gestureRecognizer.HoldCanceled += GestureRecognizer_HoldCanceled;
+
+            gestureRecognizer.ManipulationStarted += GestureRecognizer_ManipulationStarted;
+            gestureRecognizer.ManipulationUpdated += GestureRecognizer_ManipulationUpdated;
+            gestureRecognizer.ManipulationCompleted += GestureRecognizer_ManipulationCompleted;
+            gestureRecognizer.ManipulationCanceled += GestureRecognizer_ManipulationCanceled;
+
+            gestureRecognizer.SetRecognizableGestures(GestureSettings.Tap | GestureSettings.ManipulationTranslate | GestureSettings.Hold);
+
+            // We need a separate gesture recognizer for navigation, since it isn't compatible with manipulation
+            navigationGestureRecognizer = new GestureRecognizer();
+
+            navigationGestureRecognizer.NavigationStarted += NavigationGestureRecognizer_NavigationStarted;
+            navigationGestureRecognizer.NavigationUpdated += NavigationGestureRecognizer_NavigationUpdated;
+            navigationGestureRecognizer.NavigationCompleted += NavigationGestureRecognizer_NavigationCompleted;
+            navigationGestureRecognizer.NavigationCanceled += NavigationGestureRecognizer_NavigationCanceled;
+
+            if (useRailsNavigation)
+            {
+                navigationGestureRecognizer.SetRecognizableGestures(GestureSettings.NavigationRailsX | GestureSettings.NavigationRailsY | GestureSettings.NavigationRailsZ);
```

Comment:
what if the dev wants a subset of the axes? i recommend we surface gesture axis selection to the profile

Reply:
also, when switching to the navigation recognizer, the dev wil lose hold and tap recognition in this implementation

Reply:
it would.

Reply:
in this case, i would say the HTK has a bug